### PR TITLE
boards/esp32-c3-devkitM-1: Prepare for the 2.1 release

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

Remove invalid `dont-append-digest` option when loading an app and add support to the board test runner for the ESP32-C3.

### Testing Strategy

board-release-tests

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: tools/board-runner/src/esp32_c3.rs [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+        "{}/{}",
+        env::var("LIBTOCK_C_TREE").unwrap(),
+        "examples/tests/mpu_walk_region/build/rv32imac/rv32imac.0x403B0060.0x3FCC0000.tbf"
+    );
+    let mut p = esp32_c3_flash(&app).unwrap();
+
+    p.exp_string("MPU Walk Regions")?;
+    p.exp_string("Walking flash")?;
+    // p.exp_string("Will overrun")?;
+    // p.exp_string("panicked at 'Process mpu_walk_region had a fault'")?;
```
Comment (hudson-ayers) [inline tools/board-runner/src/esp32_c3.rs:276]:
```suggestion
```
No need for these since the board does not have a button attached, right?

**Review by hudson-ayers — APPROVED** [review]

File: tools/board-runner/src/esp32_c3.rs (hunk 1 above)
Comment (alistair23) [reply tools/board-runner/src/esp32_c3.rs:276]: The board has a physical button. it's not attached in Tock though

Comment (hudson-ayers) [reply tools/board-runner/src/esp32_c3.rs:276]: gotcha, makes sens to keep them then

**Review by bradjc — APPROVED** [review]

**Pull request merged**